semantic-analysis/src/Analysis/File.hs

```diff
@@ -19,3 +24,10 @@ data File a = File
 fromBody :: HasCallStack => a -> File a
 fromBody body = File (Path.absRel (srcLocFile srcLoc)) (spanFromSrcLoc srcLoc) body where
   srcLoc = snd (fromJust (listToMaybe (getCallStack callStack)))
+
+-- | The language of the provided file, as inferred by 'Language.forPath'.
+fileLanguage :: File a -> Language
+fileLanguage = Language.forPath . filePath
+
+fromPath :: Path.PartClass.AbsRel ar => Path.File ar -> File Language
+fromPath p = File (Path.toAbsRel p) lowerBound (Language.forPath p)
```

Comment:
This used to be called `fileFromTypedPath`. Since `FilePath` is now no longer part of the API, we no longer need to differentiate, and can give it a simpler name.

Reply:
Excellent!